internal/providers/gitlab/webhooksecret/whsecret.go

```diff
@@ -0,0 +1,57 @@
+//
+// Copyright 2024 Stacklok, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package webhooksecret provides a way to generate and verify secrets for GitLab webhooks.
+package webhooksecret
+
+import (
+	sum "crypto/sha512"
+	"encoding/hex"
+	"errors"
+	"fmt"
+)
+
+var (
+	// ErrEmptyBaseOrUniq is returned when the base or uniq strings are empty.
+	ErrEmptyBaseOrUniq = errors.New("base or uniq strings are empty")
+)
+
+// New creates a new secret for usage in the gitlab webhook.
+// The secret is generated by combining the base and uniq strings
+// and then hashing the result.
+func New(base string, uniq string) (string, error) {
+	if base == "" || uniq == "" {
+		return "", ErrEmptyBaseOrUniq
+	}
+
+	hash := sum.New()
+	_, err := hash.Write([]byte(base + uniq))
```

Comment:
I think this is OK for simplicity sake to bootstrap the gitlab provider, but for production-ready secret handling I think we should go further (for github also). What I would like to be able to do is to rotate the minder secrets without having to rotate the provider secrets and conversely being able to rotate provider without having to rotate the minder secret. We already had a design for this a while ago with @dmjb and I think we even have an issue open.